### DISCUSSION TITLE
Variables

### DISCUSSION THREAD

**Discussion by nc** [code-hike/codehike#270] [Q&A] [answered]
I'd like to replace $appId and $apiKey variables with an actual users ones in documentation. Is it possible to do this with Code Hike?

**Comment by pomber** [comment, marked as answer]
It's not possible right now (well, if your code doesn't change you can use the `withClass` annotation and replace the content inside a `useEffect`, it's hacky but it may work).

But it's an interesting feature, it will be useful for other use cases too. We may have support for this in the future.

**Reply by pomber** [reply]
Maybe you can do it with a css variable instead of `useEffect`. See this [playground](https://play.codehike.org/#N4IgtgJgHiBcIB0B2BiABACQKYBscHtlkBRCASwBc0wsA6IpAA2YCsBnZAek7QHdKAFgGEcAQzZsA2gEYAbAF00ABzEBjLAPw4IWAE7JV+JGyqilZANJYAnmgC8aBIkRNmDEABoQqiXBC0lXSwANzIsXgBaQyQKUTIkPTRgRyQ0alFdAHN42DQADiUoAG4UtIiIigEsCLMyCIBrG1ynaQAmAGYAFgBWWSdkAF8GALUNLR1dWFhRADMKROBkNOj5mNzgjIAKcsrq2oabAEoipbRDAkm0TKCsJGtcAl4TpAHPbyMZsky4UBx4rAAcgBXMAAIz0bDgM1EODYWC8bE0vCE+CU1gAQkCKBQjFCYXCvLsaH4wKJ5royDCIhAMo1dCABgMgA).

**Reply by pomber** [reply]
the problem with using `::after` is that it won't be copypastable, as far as I know.

**Reply by nc** [reply]
I can probably override ```document.addEventListener('copy', function(e){
  var text = window.getSelection().toString().replace(/[\n\r]+/g, '');
  e.clipboardData.setData('text/plain', text);
  e.preventDefault();
});``` the copy event like this.

**Comment by nc** [comment]
Oh wild. What a neat hack.
ᐧ
